app/code/Magento/Inventory/Model/Source/Validator/PreventDisableValidator.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Inventory\Model\Source\Validator;
+
+use Magento\Framework\Validation\ValidationResult;
+use Magento\Framework\Validation\ValidationResultFactory;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\InventoryCatalogApi\Api\DefaultSourceProviderInterface;
+
+/**
+ * Check that default source always enabled
+ */
+class PreventDisableValidator implements SourceValidatorInterface
+{
+    /**
+     * @var ValidationResultFactory
+     */
+    private $validationResultFactory;
+
+    /**
+     * @var DefaultSourceProviderInterface
+     */
+    private $defaultSourceProvider;
+
+    /**
+     * @param ValidationResultFactory $validationResultFactory
+     * @param DefaultSourceProviderInterface $defaultSourceProvider
+     */
+    public function __construct(
+        ValidationResultFactory $validationResultFactory,
+        DefaultSourceProviderInterface $defaultSourceProvider
+    ) {
+        $this->validationResultFactory = $validationResultFactory;
+        $this->defaultSourceProvider = $defaultSourceProvider;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function validate(SourceInterface $source): ValidationResult
+    {
+        $sourceCode = (string)$source->getSourceCode();
+        if ($sourceCode != $this->defaultSourceProvider->getCode()) {
```

Comment:
use strict comparison !==

Reply:
Sure, no problem, I'll do that right now